src/object/obj_tx.c

```diff
@@ -1753,7 +1753,9 @@ dc_tx_commit_prepare(struct dc_tx *tx, tse_task_t *task)
 	}
 
 	req_cnt = tx->tx_read_cnt + tx->tx_write_cnt;
+	D_RWLOCK_RDLOCK(&tx->tx_pool->dp_map_lock);
 	tgt_cnt = pool_map_target_nr(tx->tx_pool->dp_map);
+	D_RWLOCK_UNLOCK(&tx->tx_pool->dp_map_lock);
```

Comment:
I'm wondering why we don't need locking in other places?

Reply:
When you "grep dp_map", you'll see that get_tgt_rank() in src/container/cli.c accessed dp_map without locking. And I'm wondering if we have similar issue with ds_pool->sp_map?